ci/main.go

```diff
@@ -0,0 +1,211 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+
+	"dagger.io/dagger"
+)
+
+const (
+	API_VERSION = "v1"
+
+	Go2WASM   = "go1.21"
+	GO2HOST   = "go1.20.4"
+	GO2PLUGIN = "go1.20.4"
+
+	// EXTRA ARGS FOR BUILDING (placed after the "go build")
+	// use -x for more details from a go compiler
+	EXTRA_HOST_ARGS = ""
+
+	SYSCALL_CLIENT_SIDE = "apiwasm/syscall/*.go"
+
+	REP = "g/file/" + API_VERSION + "/file.pb.go"
+)
+
+var (
+	// protobuf files
+	apiProto  string
+	testProto string
+
+	// protoc plugin
+	template    string
+	eneratorSrc string
+)
+
+func main() {
+	if err := build(context.Background()); err != nil {
+		log.Fatalf("Cannot build dagger pipeline %v\n", err)
```

Comment:
This is a good construct --  test for error and use log.Fatalf to stop -- when used with command line programs.